### ISSUE TITLE
Extensions for supervised learning 

### ISSUE THREAD

**Issue by schaul** [pybrain/pybrain#20]
Thomas:

> I also think our supervised trainers are not flexible enough. AFAIK, the mean squared error function is hard-coded, the other day we needed a cross-entropy error and it wasn't apparent how to do that. 
> 
> And regularization would be nice, in a general form (not just weight decay as option for the trainer). What about conjugate gradient etc?

Martin:

> I totally agree with Thomas here (being mostly interested in supervised learning myself). The old-fashioned supervised stuff seems to be quite inflexible so far; I'm not even sure if second order methods like CG or more complicated error functions would fit into the current framework.

**Comment by brandonrobertz** [comment]
I realize this is old, but I had to change the loss function and was required to alter backprop.py. There's no way to do this modularly as the loss function is hardcoded: https://github.com/pybrain/pybrain/blob/master/pybrain/supervised/trainers/backprop.py#L95

**Comment by gregvds** [comment]
No news for a CG trainer? I'm interested :-)...
